### ISSUE TITLE
Mandatory/advisory cycle lane

### ISSUE THREAD

**Issue by mvl22** [cyclestreets/tflcid-conversion#22]
Mike Baggaley wrote in:
https://lists.openstreetmap.org/pipermail/talk-gb/2019-September/023521.html

I would prefer not to see `cycleway:lane=mandatory` as this suggests that cyclists have to use the lane when set. In the UK, the significance of the solid white line separating a cycle lane and main carriageway is that motor vehicles are not allowed to use the cycle lane, but cyclists can use either the cycle lane or main carriageway. I would only want to see mandatory used if there is also a separate sign prohibiting cyclists from the road (and I am not sure whether any of these exist). I suggest `cycleway:lane={exclusive|advisory}` which are existing tags according to the wiki. Note that UK cycle lanes can also be used by pedestrians, so are not strictly exclusive to cyclists.

> • Mandatory/Advisory Cycle Lane: OSM has no differentiation between mandatory (solid white line) and advisory (dashed white line) lane, probably because this distinction is rare elsewhere in the world. A new tag cycleway:lane={mandatory|advisory} is proposed as a backwards-compatible addition that elaborates on cycleway=lane. This would be useful for routing engines, who could infer a level of commitment to cyclists at each such location.
> https://bikedata.cyclestreets.net/tflcid/conversion/#clt_mandat

**Comment by mvl22** [comment]
Richard Fairhurst wrote in:
https://lists.openstreetmap.org/pipermail/talk-gb/2019-September/023522.html

"Strong preference _against_ this. "mandatory" is UK highway-speak but likely to be very counterintuitive to anyone who doesn't know what TSRGD stands for. To the non-initiate, it implies that use of the cycleway is mandatory for cyclists (a la `bicycle=use_sidepath`), which it isn't.

There was a long thread about this over on forum.osm.org: https://forum.openstreetmap.org/viewtopic.php?id=61427 . My preference is to map the meaning ("cars are/aren't allowed in the bike lane") by using traditional access tags: `cycleway:motor_vehicle=no, cycleway:motor_vehicle=yes`. But have a read of the thread and see what you think."

**Comment by mvl22** [comment]
Wulf4096 wrote in:
https://lists.openstreetmap.org/pipermail/talk-gb/2019-October/023591.html

"Hello,
in Germany we've got "Radfahrstreifen" (solid line) which are additionally marked by bicycle signs. Only cyclists may use those, and the sign forbids cyclists to use the main carriageway, unless they've got a reason to.

And we've got "Schutzstreifen" (dashed line). Legally, from the view of a cyclist, those marking don't exist, as they don't impose any rules on cyclists (this has been ruled by court).  Other traffic may not use the dashed lanes unless they've got a reason to.

So I guess it's similar markings and rules?

https://wiki.openstreetmap.org/wiki/DE:Bicycle/Radverkehrsanlagen_kartieren
When tagging, the only distinction is in `cycleway:right:traffic_sign`: `DE:237` vs. `DE:340`. That needs better tagging in my opinion.

I strongly object against using words like "mandatory" or "compulsory" when tagging: Nobody is legally required to use those lanes. One can always mutate to a pedestrian, turn around, or find another way. What it really means is: "Do not use the main carriageway unless you have to". And for this we've already got a widely used tag: `bicycle=use_sidepath`. Currently the tag is only used where OSM has two separate ways, but I don't see any good reason not to tag `bicycle=use_sidepath` + `cycleway:right=lane` on the same way.

This still doesn't distinguish if and how *other* traffic may use those lanes. `cycleway:*:lane={exclusive|advisory}` sounds like a reasonable tagging scheme to me:
https://wiki.openstreetmap.org/wiki/Key%3Acycleway%3Alane

tl;dr:
Use `cycleway:*:lane={exclusive|advisory}` as suggested by Mike, and one may use `bicycle=use_sidepath` to state that cycling on the main carriageway should be avoided."

**Comment by mvl22** [comment]
Andy Allan wrote in:
https://lists.openstreetmap.org/pipermail/talk-gb/2019-October/023592.html

> So I guess it's similar markings and rules?

"Similar markings, but different rules. There is no implication in the UK that a cyclist has to use a cycle lane, regardless of markings. You can ignore a solid-line-lane and ride in any other traffic lane, without needing any reason."

**Comment by mvl22** [comment]
Phillip Barnett wrote in:
https://lists.openstreetmap.org/pipermail/talk-gb/2019-October/023593.html

"Apart from on motorways!"

**Comment by mvl22** [comment]
Twitter user "ParisBeauAVelo" wrote at:
https://twitter.com/ParisBeauAVelo/status/1179512172696748032

"Mandatory/advisory cycle lane
We have this scheme
cycleway=lane + cycleway:lane=exclusive
https://wiki.openstreetmap.org/wiki/Key:cycleway:lane "

**Comment by b33k34** [comment]
Since most motorists neither understand nor behave differently around painted advisory or mandatory lanes I'm not sure there is any benefit in recording them as different categories.  I'm not aware of any evidence of either  subjective or objective safety.

**Comment by mvl22** [comment]
That may be the case, but there is a legal difference between these. Parking, for instance, is not legally permitted in a mandatory lane, whereas an advisory lane does permit this.

OSM aims to represent correctly what is on the ground, so this is a valid difference to show.

**Comment by rskedgell** [comment]
When this was originally discussed, there may have been less value in importing whether a lane is advisory or mandatory (=exclusive). However, there have been two recent legislative changes which make the distinction more useful, as mandatory lanes may be less likely to be obstructed by motor vehicles.

Firstly, the law changed in England to allow CCTV parking enforcement to be used where a mandatory cycle lane also has waiting restrictions (yellow lines) and/or loading restrictions (yellow kerb blips).
The Civil Enforcement of Parking Contraventions (England) General (Amendment) Regulations 2020
https://www.legislation.gov.uk/uksi/2020/548/made

Secondly, TfL now have powers to use CCTV for civil enforcement of the moving traffic offences committed by driving in a mandatory cycle lane or a cycle track.
https://tfl.gov.uk/info-for/media/press-releases/2022/june/tfl-to-begin-using-new-enforcement-powers-to-improve-cycle-safety-on-its-roads

**Comment by mvl22** [comment]
> When this was originally discussed, there may have been less value in importing whether a lane is advisory or mandatory (=exclusive). However, there have been two recent legislative changes which make the distinction more useful, as mandatory lanes may be less likely to be obstructed by motor vehicles.

Yes, enablement of the TMA 2004 pt. 6 now enables local authorities _outside London_ to apply for new powers to enforce mandatory cycle lanes and other things using camera enforcement:

https://www.gov.uk/government/publications/traffic-management-act-2004-summary/traffic-management-act-2004-summary#civil-enforcement-of-traffic-contraventions-tma-2004-part-6

https://www.camcycle.org.uk/magazine/magazine152/article16/

**Comment by mvl22** [comment]
> I suggest cycleway:lane={exclusive|advisory} which are existing tags according to the wiki.

Agreed that these are the correct tags to be using now, as per the [Wiki entry](https://wiki.openstreetmap.org/wiki/Key:cycleway:lane).

[Taginfo shows](https://taginfo.openstreetmap.org/keys/cycleway:lane#values) that `advisory` and `exclusive` are now in accepted use.

**Comment by mvl22** [comment]
@systemed

The action here is to add to [readers/clt_roads.rb](https://github.com/cyclestreets/tflcid-conversion/blob/master/readers/clt_roads.rb) the following mapping:

- If `CLT_MANDAT` is '`TRUE`' add `cycleway:lane=exclusive`
- If `CLT_ADVIS` is '`TRUE`' add `cycleway:lane=advisory`
